reconcile/checkpoint.py

```diff
@@ -0,0 +1,123 @@
+"""Performs an SRE checkpoint.
+
+The checks are defined in
+https://gitlab.cee.redhat.com/app-sre/contract/-/blob/master/content/process/sre_checkpoints.md
+
+"""
+import logging
+import re
+from functools import partial
+from http import HTTPStatus
+from pathlib import Path
+from typing import Any, Iterable, Mapping, Union
+
+import requests
+from jinja2 import Template
+from jira import Issue
+
+from reconcile.utils.constants import PROJ_ROOT
+from reconcile.utils.jira_client import JiraClient
+
+
+DEFAULT_CHECKPOINT_LABELS = ('sre-checkpoint',)
+
+# We reject the full RFC 5322 standard since many clients will choke
+# with some carefully crafted valid addresses. We might
+EMAIL_ADDRESS_REGEXP = re.compile(r'^\w+[-\w\.]*@(?:\w[-\w]*\w\.)+\w+')
+MAX_EMAIL_ADDRESS_LENGTH = 320  # Per RFC3696
+
+MISSING_DATA_TEMPLATE = PROJ_ROOT / 'templates' / \
+    'jira-checkpoint-missinginfo.j2'
+
+
+def url_makes_sense(url: str) -> bool:
+    """Guesses whether the URL may have a meaningful document.
+
+    Obvious cases are if the document can be fully downloaded, but we
+    also accept that the given document may require credentials that
+    we don't have.
+
+    The URL is non-sensical if the server is crashing, the document
+    doesn't exist or the specified URL can't be even probed with GET.
+    """
+    # TODO: we really should verify SSL certificates here. We
+    # shouldn't let important service information in shady self-signed
+    # servers
+    rs = requests.get(url, verify=False)
+    # Codes above NOT_FOUND mean the URL to the document doesn't
+    # exist, that the URL is very malformed or that it points to a
+    # broken resource
+    return (rs.status_code < HTTPStatus.NOT_FOUND)
+
+
+def valid_owners(owners: Iterable[Mapping[str, str]]) -> bool:
```

Comment:
💡 we could have a look at an identity store (LDAP, something else?) to figure out if the mail address exists and belongs to a person that still works for the company.